### DISCUSSION TITLE
Readability and regex

### DISCUSSION THREAD

**Discussion by problemadder** [ObsidianToAnki/Obsidian_to_Anki#206] [Q&A]
To create notes that look good and have a good readability within obsidian regex is key.
Obsidian Notes, that look like this are ugly:

START
Basic
Question
Answer: Answer Tags: test
END

But regex is quite hard for beginners like me. I would love to share ideas and see implementations of customized regex variants.

**Comment by problemadder** [comment]
A starting point would be a HOW-TO add an anki field (like "Extra:" or "Hint") to an existing regex?
Has somebody done this? Perhaps the "Question answer style" would be a good starting point.
Instead of the format mentioned above, this would be better and more readable within obsidian:

Q: "Question"
A: "Answer"
E: "Extra contents, like example an sentence"

**Comment by uuuser-name** [comment]
Hi.
Great discussion idea.
For the Basic card type, my current syntax is `question::answer::context::example`. 
I modified the anki fields accordingly, adding a "context" field and an "example" field.
For the Basic (and reversed) card type, I use the same syntax but with 3 colons separating the fields, like `question:::def:::context:::example`.

I think people exchanging their syntaxes and advice could lead to a great discussion.

**Reply by problemadder** [reply]
do you mind showing me your regex for this one. would like to try it. I love the "::" and ":::" because the cards look good in obsidian too. But on the other hand cards with clozed deletion like this {missing word::hint for missing word} dont work with regex and the "START END" cards are kind of ugly in obsidian. 
Currently I use "::" for vocabulary (Basic and reversed) and the Q: A: type for other stuff. I start to think that a KISS "Keep it Short and Stupid" approach -> Cards with simple Front and Back info are the best.

**Comment by RolandSherwin** [comment]
you can put them inside a hidden paragraph like 

```
<p hidden>
START
Basic
Question
Back: </p>
Answer
<p hidden>
END </p>
```

Thus only the "Answer" will be visible in preview mode.

**Comment by weriton1** [comment]
I know Im quite late to the discussion here but i'd really recommend learning RegEx using [Regexone](https://regexone.com/). That's how I learnt how to use it. It's a great site that goes through the basic functions one by one.

**Comment by MotaOcimar** [comment]
I strongly agree that readability is a key feature.
An idea to improve this is to use specific rules for `START` and `END`.
Also specify the card type in a html comment so that it does not appear in Obsidian preview.

Example:

    ---
    
    |This is the front of the basic card|
    |-----------------------------------|
    |This is the back of the basic card |
    ***
    
    ---
    
    This is a ==cloze== card
    ***

It will look like this in the Obsidian preview:

![image](https://user-images.githubusercontent.com/29007115/137607393-0cc189a4-73f0-45f5-a54f-1b56b63e78a9.png)

It is needed to implement it, of course
